### DISCUSSION TITLE
Having trouble accessing component fields through another page's relationship field.

### DISCUSSION THREAD

**Discussion by DavidZudu** [flyntwp/flynt#404] [Q&A] [answered]
1. The component "BlockSummary" is added to a custom post type post labelled "Example Post"
2.  Home page adds component "BlockPostSelector" which has a relationship field for selecting posts with the intention to display specific content from it.
3. "Example Post" is selected in this relationship field.
4. In the "BlockPostSelector" component twig, I need to display content from a field in "BlockSummary" used on "Example Post".

How do I do this? I can create a for loop for looping through selected posts, but can only display things like title, slug etc, not a field from a component used on the page.

`
{% for post in relationshipField %}
 {{ post.title }} // this works
 {{ post.summaryText }} // this does not display the field from the BlockSummary
{% endfor %}
`

This seems like it should be easy but trying to describe it is a bit complex. Thanks for any help.

**Comment by aaronmeder** [comment]
Hmm without being sure can you try to use `{{ post.meta('summaryText') }}` ?

Btw: I often find the [Timber docs](https://timber.github.io/docs/reference/timber-post/#meta) quite helpful regarding issues like this.

**Reply by DavidZudu** [reply]
Unfortunately this doesn't appear to work. Is it something to do with the field technically being a sub field of a Flexible Content field? I can't see in the docs a way to reference a sub field to test if this is the case.

The twig/Timber docs help with a lot of things but there's a few things that seem to be Flynt specific that I struggle to find answers to.

**Comment by DavidZudu** [comment, marked as answer]
Ok I figured it out. For Flynt specifically, its likely your fields are technically inside Flexible Layout fields, so you need to loop through your Flexible Layout and find the correct Components, then you can reference the fields: [here in the docs](https://timber.github.io/docs/guides/acf-cookbook/#flexible-content-field)

``` 
{% for post in relationshipField %}  // get the selected posts from the relationship/post object fields
    {{ post.title }} // This works like you would expect
    {% for content_block in pageComponents %} // Loop though each layout block used in you flexible layout
        {% if content_block.acf_fc_layout == 'blockSummary' %} // Getting the correct layout
            {{content_block.summaryText}}  // It works!
        {% endif %}
    {% endfor %}
{% endfor %}
```
Extra loops to jump through (literally?) but glad it works.

**Reply by aaronmeder** [reply]
Ah that would kina make sense @DavidZudu.
I realise that I actually always added "normal ACF fields" for that kind of data to my CPTs, which you could call using `post.meta` if I remember it right. Think I never tried grab data from an other posts pageComponents.

One note maybe: you actually cannot be sure that these components are in the target post because these can be flexibly chosen...by adding them as normal ACF fields (through the normal ACF fieldset options) you can make them mandatory.

**Reply by marcotiberio** [reply]
Hi @DavidZudu I'm glad I've stumbled on your fix for this, I was looking for the same issue. However, now I'm trying to use the fix on a post_object component but I can't get through it. This is what I have. Do you maybe have any inisght on this? Thanks!
```
{% for post in post %} // "post" is the name of the post_object component on the homepage
   {% for content_block in artistComponents %} // "artistComponents" is postComponents but for a specific CPT
      {% if content_block.acf_fc_layout == 'BlockArtistInfo' %} // "BlockArtistInfo" is the name of the component I wan't to access
         {{ content_block.bioHtml|e('wp_kses_post') }} // "bioHtml" is the name of the field I want to get
      {% endif %}
   {% endfor %}
{% endfor %}
```

**Reply by comsup** [reply]
I had the same issue on a posts list page. The problem is if you have multiple flexible content on a post page you can't automatically retrieve them on an archive page by example.
What I did and it works is to generate an array of ACF data on your functions.php, and after you can get them on a loop on your twig template : 

```
$allPost = new PostQuery();
 $allAcfPostsData = array();
foreach ($allPost as $post){
  $data_acf = get_fields( $post->ID );
  $allAcfPostsData[$post->ID] = $data_acf;
}
$data['allAcfPostsData'] = $allAcfPostsData;
return $data;

{% for content_block in allAcfPostsData[post.id].postComponents %}
  {% if content_block.acf_fc_layout == 'ComsUpBlockTitrePage' %}
    {{content_block.h1Page}}
  {% endif %}
{% endfor %}
```
